pkgs/tools/graphics/escrotum/default.nix

```
@@ -2,7 +2,8 @@
 }:
 
 with python2Packages; buildPythonApplication {
-  name = "escrotum-2019-06-10";
+  pname = "escrotum";
+  version = "unstable-2019-06-10";
```

This changes the package name.

AFAIK, Nix still ignores pname and version fields and parses the version from the name field. For Nix, versions always start with the number, so if you use `version = "unstable-2019-06-10";`, the Nix will show name as `escrotum-unstable` and version as `2019-06-10`.

It’s better to make this explicit:
```suggestion
  pname = "escrotum-unstable";
  version = "2019-06-10";
```

On the other hand, if we want to use pname & version to generate package list for Repology, the pname should stay “escrotum”.

---

my understanding was:
if `name` is present use it, parse it, ...
else look for `pname` and `version`
if i look at the [package search](https://search.nixos.org/packages?channel=21.05&show=escrotum&from=0&size=50&sort=relevance&query=escrotum) there is a breaking change if i would have the `-unstable` in the `pname`.

---

If that didn’t change, Nix does not know about `pname`/`version`—it only knows `name`.

If you look at mkDerivation, the `meta.name` is always set from either `name` or `${pname}-${version}`: https://github.com/NixOS/nixpkgs/blob/0a1dca4bf2cb2ca77830f6bddcc790cae7d314e7/pkgs/stdenv/generic/make-derivation.nix#L340

The top-level `name` is set similarly from `name` or `pname`-`version`: https://github.com/NixOS/nixpkgs/blob/3e49dac3e4d8b9f7f9cf887e73c31e56062b8333/pkgs/stdenv/generic/make-derivation.nix#L202-L223

So for Nix it does not matter whether you put “unstable” in pname or version—they’ll get concatenated and parsed anyway.

For package search, it’ll be a breaking change either way. The good news is that the `nix-env -iA` command does not use name at all—it’s an attribute name in `all-packages.nix` and that does not change.

---

thanks nice to know.
is your suggestion to remove the `unstable` part?

---

I’m not sure what I am suggesting :slightly_smiling_face: 

Ideally, I would append `-unstable` to `pname` (because it _is_ a part of package name as far as Nix is concerned).

However, if pname/version are used for repology, then adding `-unstable` to `pname` would create a mismatch between nixpkgs name and the rest of distros.

Grepping through nixpkgs, it looks like adding “unstable” to the version is more common, so it’s probably fine to leave it as is. I’ll mark this conversation as resolved. Sorry for the false alarm